tests/testthat/test-unite.R

```diff
@@ -71,3 +71,14 @@ test_that("validates its inputs", {
     unite(df, "z", x:y, na.rm = 1)
   })
 })
+
+test_that("unite works for empty selections", {
+  variables <- c()
+  df <- tibble(
+    x = c("x", "y", "z"),
+    y = 1L
+  )
+  expect_no_error(df2 <- unite(df, col = "new_col", all_of(variables)))
+  expect_identical(colnames(df2), c("x", "y", "new_col"))
+  expect_identical(unique(df2$new_col), "")
+})
```

Comment:
IIUC, this case with 0 column data frames should also work now

```r
library(tidyr)

df <- tibble(.rows = 2L)

unite(df, "foo")
#> Error in `df_append()`:
#> ! `after` must be a whole number, not an integer `NA`.
#> ℹ This is an internal error that was detected in the tidyr package.
```

I think we expect to see 1 new column named `"foo"` with 2 rows of `""` values

Can you please add a test for that too?